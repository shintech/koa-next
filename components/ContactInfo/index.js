import PropTypes from 'prop-types'
import Wrapper from './wrapper'

const ContactInfo = ({ email = 'example@example.org' }) =>
  <Wrapper>
    <a href={`mailto:${email}`}>{email}</a>
  </Wrapper>

ContactInfo.propTypes = {
  email: PropTypes.string.isRequired
}

export default ContactInfo
