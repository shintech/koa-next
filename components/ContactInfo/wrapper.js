import styled from 'styled-components'

const Wrapper = styled.div`
  a {
    font-size 4ch;
    color: lightgreen;
    text-decoration: none;
    text-shadow: 0.2ch 0.2ch black;
  }
`

export default Wrapper
