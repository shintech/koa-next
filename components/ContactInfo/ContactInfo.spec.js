/* eslint-env jest */

import { shallow } from 'enzyme'
import { compose } from 'redux'
import toJSON from 'enzyme-to-json'
import ContactInfo from 'components/ContactInfo'

const shallowExpect = compose(expect, toJSON, shallow)

describe('COMPONENT -> ContactInfo - snapshot...', () => {
  it('expect to render correct properties', () => {
    shallowExpect(<ContactInfo email='email@example.org' />).toMatchSnapshot()
  })
})
