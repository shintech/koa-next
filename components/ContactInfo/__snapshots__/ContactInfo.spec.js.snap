// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT -> ContactInfo - snapshot... expect to render correct properties 1`] = `
<wrapper__Wrapper>
  <a
    href="mailto:email@example.org"
  >
    email@example.org
  </a>
</wrapper__Wrapper>
`;
